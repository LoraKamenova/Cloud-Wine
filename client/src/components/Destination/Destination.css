.destination-page {
    background-image: url('https://res.cloudinary.com/dyhmxus4n/image/upload/v1615655436/React%20Project/wine_map_p79f5c.jpg');
    height: 100vh;
}

.destination-section {
    margin-right: 7%;
}

.destination-page > .general-pages-wrapper > .destination-section {
    height: 480px;
    margin-top: 60px;
}

.destination-page > .general-pages-wrapper > Aside > Footer > .footer {
    color: black;
}
