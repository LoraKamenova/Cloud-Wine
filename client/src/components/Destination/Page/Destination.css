.destination-image-wrapper {
    background-image: url('https://res.cloudinary.com/dyhmxus4n/image/upload/v1615655436/React%20Project/wine_map_p79f5c.jpg');
}

.destination-section > .destination-content-container > .destination-article-wrapper > .info-button-wrapper > a {
    text-decoration: none;
}

@media all and (max-width: 1200px) {
    .destination-image-container {
        display: none;
    }
    .destination-article-wrapper {
        width: 100%;
        height: auto;
        padding: 55px;
    }

    .info-button-wrapper {
        justify-content: center;
        margin-top: 20px;
    }
}
