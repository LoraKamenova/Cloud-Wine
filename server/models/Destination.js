const mongoose = require('mongoose');
const { String } = mongoose.Schema.Types;

const DestinationSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: true,
    },
    address: {
        type: String,
        required: true,
    },
    lat: {
        type: String,
        required: true,
    },
    lon: {
        type: String,
        required: true,
    },
    services: {
        type: String,
        required: true,
    },
    logoUrl: {
        type: String,
        required: true
    },
    imageUrl1: {
        type: String,
        required: true
    },
    imageUrl2:{
        type: String,
        required: true
    },
    mapUrl: {
        type: String,
        required: true,
    },
});

module.exports = mongoose.model('Destination', DestinationSchema);
