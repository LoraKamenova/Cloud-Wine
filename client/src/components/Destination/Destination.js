import './Destination.css'
import Article from "../Article/Article";
import {Link} from "react-router-dom";
import SubArticle from "../SubArticle/SubArticle";

const Destination = () => {
    return (
        <section className="custom-section destination-section">
            <div className="content-container">
                <div className="upper-part-container">
                    <div className="article-wrapper">
                        <Article
                            title="ДЕСТИНАЦИИ"
                            paragraph="Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                            Phasellus pulvinar tellus nec ante volutpat, vel commodo nisl aliquam. Maecenas nec
                            malesuada risus, ut maximus libero. Aliquam bibendum sollicitudin nibh, ac vehicula leo
                            maximus vitae. Aenean vitae metus luctus, molestie leo ac, ullamcorper justo. Class aptent
                            taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Mauris vitae arcu orci.
                            Suspendisse potenti. Sed non maximus est. Phasellus vel sem molestie, sollicitudin sapien
                            ac, condimentum erat. Etiam quis augue aliquet, laoreet risus vel, lacinia orci. Fusce vitae
                            congue sapien. Curabitur sed lacus ut dolor auctor fermentum ac sed risus. Aliquam quis risus vel felis cursus
                            eleifend ac a enim. Curabitur dapibus diam sit amet scelerisque ultricies."
                        />
                    </div>
                    <div className="image-container">
                        <div className="image-wrapper destination-image-wrapper"></div>
                    </div>
                </div>
                <div className="lower-part-container">
                    <SubArticle
                        paragraph="Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                            Phasellus pulvinar tellus nec ante volutpat, vel commodo nisl aliquam. Maecenas nec
                            malesuada risus, ut maximus libero. Aliquam bibendum sollicitudin nibh, ac vehicula leo
                            maximus vitae. Aenean vitae metus luctus, molestie leo ac, ullamcorper justo. Class aptent
                            taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Mauris vitae arcu orci.
                            Suspendisse potenti. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                            Phasellus pulvinar tellus nec ante volutpat sto tellus nec ante volutpat sto. "
                    />
                    <div className="info-button-wrapper">
                        <Link to="/all-destinations"><span className="button info-button">Виж дестинациите<i
                            className="fas fa-angle-double-right"></i></span></Link>
                    </div>
                </div>
            </div>
        </section>

    );
};

export default Destination;
