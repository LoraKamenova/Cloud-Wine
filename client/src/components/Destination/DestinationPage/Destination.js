import './Destination.css'
import Article from "../../Other/Article/Article";
import { Link } from "react-router-dom";

const Destination = () => {
    return (
        <section className="custom-section destination-section">
            <div className="destination-content-container">
                    <div className="destination-article-wrapper">
                        <Article
                            title="ДЕСТИНАЦИИ"
                            paragraph="Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                            Phasellus pulvinar tellus nec ante volutpat, vel commodo nisl aliquam. Maecenas nec
                            malesuada risus, ut maximus libero. Aliquam bibendum sollicitudin nibh, ac vehicula leo
                            maximus vitae. Aenean vitae metus luctus, molestie leo ac, ullamcorper justo. Class aptent
                            taciti sociosqu ad litora torquent per conubia nostra, per"
                        />
                        <div className="info-button-wrapper">
                            <Link to="/destination/all"><span className="button info-button">Виж дестинациите<i className="fas fa-angle-double-right"></i></span></Link>
                        </div>
                    </div>
                    <div className="destination-image-container">
                        <div className="destination-image-wrapper"></div>
                    </div>
                </div>
        </section>
    );
};

export default Destination;
