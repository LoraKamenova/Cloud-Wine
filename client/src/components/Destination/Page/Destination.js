import './Destination.css'
import AsideArticle from "../../Shared/AsideArticle/AsideArticle";

import { Link } from "react-router-dom";

const Destination = () => {
    return (
        <section className="destination-section">
            <div className="destination-content-container">
                    <div className="destination-article-wrapper">
                        <AsideArticle
                            title="ДЕСТИНАЦИИ"
                            paragraph="България предлага разнообразни възможности за винен туризъм. Според личните ви предпочитания можете да се разходите из лозови масиви, да разгледате винарска изба и, разбира се, да дегустирате различни вина. Голяма част от винарски изби имат собствени ресторанти, а някои разполагат и с хотелска част, където може да отседнете."
                        />
                        <div className="info-button-wrapper">
                            <Link to="/destination/all"><span className="button info-button">Виж дестинациите<i className="fas fa-angle-double-right"></i></span></Link>
                        </div>
                    </div>
                    <div className="destination-image-container">
                        <div className="destination-image-wrapper"></div>
                    </div>
                </div>
        </section>
    );
};

export default Destination;
