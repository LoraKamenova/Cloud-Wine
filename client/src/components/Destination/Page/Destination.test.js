import { render } from '@testing-library/react';
import Destination from './Destination';
import {BrowserRouter} from "react-router-dom";

describe('Destination Component', () => {

    it("Destination Component snapshot", () => {
        const { asFragment } = render(<BrowserRouter><Destination /></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

});
