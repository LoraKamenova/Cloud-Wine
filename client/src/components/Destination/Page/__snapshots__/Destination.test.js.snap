// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Destination Component Destination Component snapshot 1`] = `
<DocumentFragment>
  <section
    class="destination-section"
  >
    <div
      class="destination-content-container"
    >
      <div
        class="destination-article-wrapper"
      >
        <article
          class="article"
        >
          <div
            class="article-heading-wrapper"
          >
            <span
              class="article-heading"
            >
              ДЕСТИНАЦИИ
            </span>
          </div>
          <p
            class="article-paragraph"
          >
            България предлага разнообразни възможности за винен туризъм. Според личните ви предпочитания можете да се разходите из лозови масиви, да разгледате винарска изба и, разбира се, да дегустирате различни вина. Голяма част от винарски изби имат собствени ресторанти, а някои разполагат и с хотелска част, където може да отседнете.
          </p>
        </article>
        <div
          class="info-button-wrapper"
        >
          <a
            href="/destination/all"
          >
            <span
              class="button info-button"
            >
              Виж дестинациите
              <i
                class="fas fa-angle-double-right"
              />
            </span>
          </a>
        </div>
      </div>
      <div
        class="destination-image-container"
      >
        <div
          class="destination-image-wrapper"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
